src/tests/unit/test_tree_functionality/test_tree_node_to_dict.py

```diff
@@ -0,0 +1,213 @@
+            entity=nested_2_data,
+            blueprint_provider=BlueprintProvider.get_blueprint,
+            parent=nested,
+            attribute=BlueprintAttribute(name="", attribute_type="Bush"),
+        )
+
+        nested_2_reference_data = {"_id": "2", "name": "Reference", "description": "", "type": "Garden"}
+        Node(
+            recipe_provider=mock_storage_recipe_provider,
+            key="reference",
+            uid="2",
+            entity=nested_2_reference_data,
+            blueprint_provider=BlueprintProvider.get_blueprint,
+            parent=nested_2,
+            attribute=BlueprintAttribute(name="", attribute_type="Garden"),
+        )
+
+        list_data = {"name": "List", "type": "Bush"}
+        list_node = ListNode(
+            recipe_provider=mock_storage_recipe_provider,
+            key="list",
+            uid="",
+            entity=list_data,
+            parent=root,
+            blueprint_provider=BlueprintProvider.get_blueprint,
+            attribute=BlueprintAttribute(name="", attribute_type="Bush"),
+        )
+
+        item_1_data = {"name": "Item1", "description": "", "type": "Garden"}
+        item_1 = Node(
+            recipe_provider=mock_storage_recipe_provider,
+            key="0",
+            uid="",
+            entity=item_1_data,
+            blueprint_provider=BlueprintProvider.get_blueprint,
+            parent=list_node,
+            attribute=BlueprintAttribute(name="", attribute_type="Garden"),
+        )
+
+        actual_root = {
+            "_id": "1",
+            "name": "root",
+            "description": "",
+            "type": "all_contained_cases_blueprint",
+            "nested": {
+                "name": "Nested",
+                "description": "",
+                "type": "Garden",
+                "nested": {
+                    "name": "Nested",
+                    "description": "",
+                    "type": "Bush",
+                    "reference": {"_id": "2", "name": "Reference", "description": "", "type": "Garden"},
+                },
+            },
+            "list": [{"name": "Item1", "description": "", "type": "Garden"}],
+        }
+
+        self.assertEqual(actual_root, tree_node_to_dict(root))
+
+        actual_nested = {
+            "name": "Nested",
+            "description": "",
+            "type": "Garden",
+            "nested": {
+                "name": "Nested",
+                "description": "",
+                "type": "Bush",
+                "reference": {"_id": "2", "name": "Reference", "description": "", "type": "Garden"},
+            },
+        }
+
+        self.assertEqual(actual_nested, tree_node_to_dict(nested))
+
+        item_1_actual = {"name": "Item1", "description": "", "type": "Garden"}
+
+        self.assertEqual(item_1_actual, tree_node_to_dict(item_1))
+
+    def test_tree_node_to_ref_dict_2(self):
+        form_node = get_form_example_node()
+        form_dict = tree_node_to_ref_dict(form_node)
```

Comment:
The last two tests are actually testing tree_node_to_ref_dict, not tree_node_to_dict, so this could be split further.

Reply:
Added to the backlog